From 20ae1bdac80679b43326321cfa892e548a3e85de Mon Sep 17 00:00:00 2001
From: "H.J. Lu" <hjl.tools@gmail.com>
Date: Wed, 29 Jan 2020 10:58:30 -0800
Subject: [PATCH 08/13] x86_64/k10/popcount.asm: Prepend X86_NOTRACK to "jmp
 *%rdx"

Since K10 popcount.asm uses a trick to implment jump tables with LEA,
prepend X86_NOTRACK to "jmp *%rdx" to disable indirect branch tracking
when Intel CET is enabled.

	* mpn/x86_64/k10/popcount.asm: Prepend X86_NOTRACK to
	"jmp *%rdx".
---
 mpn/x86_64/k10/popcount.asm | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/mpn/x86_64/k10/popcount.asm b/mpn/x86_64/k10/popcount.asm
index 735d25a30..45bcba5b0 100644
--- a/mpn/x86_64/k10/popcount.asm
+++ b/mpn/x86_64/k10/popcount.asm
@@ -79,7 +79,7 @@ C	neg	R32(%rcx)
 
 	lea	L(top)(%rip), %rdx
 	lea	(%rdx,%rcx,2), %rdx
-	jmp	*%rdx
+	X86_NOTRACK jmp	*%rdx
 ',`
 	lea	(up,n,8), up
 
@@ -101,7 +101,7 @@ C	lea	(%rcx,%rcx,4), %rcx	C 10x
 
 	lea	L(top)(%rip), %rdx
 	add	%rcx, %rdx
-	jmp	*%rdx
+	X86_NOTRACK jmp	*%rdx
 ')
 
 	ALIGN(32)
-- 
2.27.0

